libkbfs/folder_branch_ops.go

```diff
@@ -2241,6 +2247,31 @@ func (fbo *folderBranchOps) doMDWriteWithRetry(ctx context.Context,
 				if err = fbo.cr.Wait(ctx); err != nil {
 					return err
 				}
+			} else if isRevisionConflict(err) {
+				// We can only get here if we are already on an
+				// unmerged branch and an errored PutUnmerged did make
+				// it to the mdserver.  Let's force sync, with a fresh
+				// context so the observer doesn't ignore the updates
+				// (but tie the cancels together).
+				newCtx := fbo.ctxWithFBOID(context.Background())
```

Comment:
Can't `newCtx := fbo.ctxWithFBOID(ctx)` work, since the new FBO ID value will hide the old one? Then you get the automatic tying of the cancels for free.